### ISSUE TITLE
Get editText field value or text

### ISSUE THREAD

**Issue by JRamos29** [joonaspaakko/ScriptUI-Dialog-Builder-Joonas#91]
Hi Joonas. 
Firstly, thanks and congratulations on this excellent work. There will be any changes to work with UXP?

So, there's some helper function to get the **editText** values generated for HTML?
I mean, when the editText is generated there's a **div** element with some **span** elements inside. 
Or should I use some jQuery or any other way of accessing them?

**Comment by joonaspaakko** [comment]
Not exactly, but at least that specific task is pretty simple:

[Codepen example](https://codepen.io/joonaspaakko/pen/dyvOoQE?editors=1010)

Vanilla JS
```js
// Edittext was given the custom name "two"
SDB.name('two').textContent
// There are those two wrapping DIV's inside the item html element, 
// but there's no need to find the inner most element in this case...
```

jQuery
```js
// You get no benefit from using jquery here, but this would be how:
$( SDB.name('two') ).text();

// or if you don't care to use SDB.name()
$('[data-item-name="two"]').text();
```

- - - - - - - - - 

It is fairly close to working in UXP as is... but not quite and I can't promise anything. The issues are all CSS related if I remember that correctly. Here's a little test I did last year https://youtu.be/2GD8b8XG47k

**Comment by JRamos29** [comment]
Yeah, it's pretty simple. I asked just in case of not finding the documentation about it.

What about the UXP, I believe that would be nice, but I agree with the fact that the SDB with CEP already is an awesome tool. And it's up to you by continuing with UXP or let that for who came after to continue with SDB. 

Anyway, keep up with the making of the great work.

**Comment by joonaspaakko** [comment]
This may come off a bit stand-offish and like a rant, but I'm just trying to lay down the main arguments against continuing with CEP/UXP support so that others may better understand why I'm hesitant and I kinda want this to be "an official statement" on my thoughts about it.

TL;DR — I'm not saying I'll never add UXP export to SDB, but right now I have no plans. I think there are better ways and tools built by people who know way more about CEP or UXP than I do, which is next to nothing... So I probably shouldn't shoehorn my tool built for another use into that space.

> Yeah, it's pretty simple. I asked just in case of not finding the documentation about it.

That's one reason I feel like I half-assed the CEP export. I chose to only make item specific helper methods if the item had some interactivity that needed to be created using JS. It was so much work already that I just decided not to include every item type. But with that... any interaction with any of the other items means that you need to kind of figure out yourself how to attach events and which child elements to target to achieve what you want. It's a little unfortunate... I don't really have a good reason why... It's because I'm very slow at making personal projects and I strongly felt this was the best I could do before they axe CEP 😄 

> What about the UXP, I believe that would be nice, but I agree with the fact that the SDB with CEP already is an awesome tool. And it's up to you by continuing with UXP or let that for who came after to continue with SDB.

I may consider UXP if there's demand, but I kinda feel like any demand would be a little misplaced. Even before I made the CEP export, I already felt like there were UI frameworks for React and Vue that would make the entire process so much easier. For CEP  this dialog builder is nothing but a blast from the past, I think, with hardly any convenience for CEP (...or UXP), other than the graphical builder you can use for the initial design. But when you add SDB's code into your CEP (or UXP) extension, it's kind of a drag to work with in comparison to just about anything, especially if that anything works with React or Vue, in which moving around components is far less of a hassle than it is in ScriptUI. In general I'm of the mind that there is no need for a graphical interface builder when it's very simple to do with a framework?

I've been saying this since I added the CEP export, but it was just an afterthought and that is the root cause it's less than ideal for anything but ScriptUI dialogs (for various reasons).

**It might make sense to just make a separate graphical interface builder for Spectrum** or something else, though that said, I was already quite out of the loop with CEP as I've only ever made one CEP extension in maybe 2014 that never went past my own desk and that ends my knowledge of CEP 😄. Actually, I'm not sure if it would make sense to make a UXP interface builder for Spectrum or any other framework at this point in time because UXP is still kinda new(?). I don't know what Adobe's plans are, but my understanding is that UXP doesn't support everything that Spectrum offers and I think it only works with React, which is a shame.

And like I said, I'm so incredibly out of the loop with CEP or UXP that I'm probably the wrongest person to make an interface builder for either. I could see myself getting involved in such a project, but I don't see myself leading such project.

Maybe @kerrishotts would be more in the loop:

1. Would it make sense to add UXP export to ScriptUI Dialog Builder?
   - Also considering it's shortcomings...
3. Is anyone working on an interface builder for CEP/UXP or thinking about it? 
4. Would it even make sense to make a completely new interface builder for Spectrum/ (CEP/UXP)?
	- SDB solved an issue with SUI, where it was difficult to move the code around in order to prototype a dialog interface... but CEP/UXP doesn't really have that issue if you use a component framework... is what I think.
	- Who would be the people to do it? I realize I don't know any CEP/UXP devs 🤨

**Comment by JRamos29** [comment]
Well, about the "official statement", I think you should put it in the SDB docs or even on the SDB page to make sure that more people or everyone that uses SDB can read. 👍 

I agree with all that you said. :

Considering that ExtendScript itself hadn't had any support or updates by Adobe (regardless CEP or now the UXP), has a very small community of developers, and a lack of resources (docs, libs, utils, etc.) over the internet, the SDB came like a great contribution (if not the greater contribution) from who want tools or to ease the scripts/extension development, mainly the UI.

In general, we see JS-based frameworks poppin' up all the time for Web / Mobile dev. But such a UI component-based tool for ExtendScript, with export options it's unique and amazing (Adobe itself, having Dreamweaver and XD, could present something like that). 😄 

About the exporting to CEP (or even UXP), as you said, it's kinda a rework for those who need to adapt it to React / Vue to work with it.

But it shows that is possible to export to CEP / UXP and perhaps even with a better approach of React/Vue components to work with ExtendScrip and the Adobe host applications.

At last but not least, congratulations on the project. 🥇 
It's the kind of tool I'd like to had built myself.  
Thanks for the inspiration, and hope that you continue with great works and soon came up with new insights and ideas. 🎉
